src/HookHandlers.php

```diff
@@ -24,6 +24,25 @@ public static function onScribuntoExternalLibraries( string $engine, array &$ext
 	 */
 	public static function onParserTestGlobals( array &$globals ): void {
 		MediaWikiServices::getInstance()->resetServiceForTesting( 'HttpRequestFactory' );
+
+		if ( version_compare( MW_VERSION, '1.41', '<' ) ) {
+			MediaWikiServices::getInstance()->redefineService(
+				'HttpRequestFactory',
+				// The below function is copied from ServiceWiring.php. Not sure how to access the original function.
+				static function ( MediaWikiServices $services ): HttpRequestFactory {
+					return new HttpRequestFactory(
+						new ServiceOptions(
+							HttpRequestFactory::CONSTRUCTOR_OPTIONS,
+							$services->getMainConfig()
+						),
+						LoggerFactory::getInstance( 'http' )
+					);
+				},
+			);
+
+			return;
+		}
```

Comment:
Annoying to have 2 big blobs like this where the only difference is the additional argument passed into HttpRequestFactory. However, something along these lines is necessary for Mw 1.39 compatibility.